src/main/java/gyro/aws/codebuild/CodebuildProjectEnvironment.java

```diff
@@ -0,0 +1,226 @@
+    }
+
+    public void setEnvironmentVariables(List<CodebuildProjectEnvironmentVariable> environmentVariables) {
+        this.environmentVariables = environmentVariables;
+    }
+
+    /**
+     * The type of credentials the build project uses to pull images in the build.
+     */
+    @Updatable
+    @ValidStrings({ "CODEBUILD", "SERVICE_ROLE" })
+    public String getImagePullCredentialsType() {
+        return imagePullCredentialsType;
+    }
+
+    public void setImagePullCredentialsType(String imagePullCredentialsType) {
+        this.imagePullCredentialsType = imagePullCredentialsType;
+    }
+
+    /**
+     * The field that specifies running the Docker daemon inside a Docker container.
+     */
+    @Updatable
+    public Boolean getPrivalegedMode() {
+        return privalegedMode;
+    }
+
+    public void setPrivalegedMode(Boolean privalegedMode) {
+        this.privalegedMode = privalegedMode;
+    }
+
+    /**
+     * The credentials for access to a private registry.
+     */
+    @Updatable
+    public CodebuildRegistryCredential getRegistryCredential() {
+        return registryCredential;
+    }
+
+    public void setRegistryCredential(CodebuildRegistryCredential registryCredential) {
+        this.registryCredential = registryCredential;
+    }
+
+    @Override
+    public void copyFrom(ProjectEnvironment model) {
+        setComputeType(model.computeTypeAsString());
+        setImage(model.image());
+        setType(model.typeAsString());
+        setCertificate(model.certificate());
+        setImagePullCredentialsType(model.imagePullCredentialsTypeAsString());
+        setPrivalegedMode(model.privilegedMode());
+
+        if (model.environmentVariables() != null) {
+            List<CodebuildProjectEnvironmentVariable> environmentVariables = new ArrayList<>();
+
+            CodebuildProjectEnvironmentVariable environmentVariable = newSubresource(
+                CodebuildProjectEnvironmentVariable.class);
+
+            for (EnvironmentVariable variable : model.environmentVariables()) {
+                environmentVariable.copyFrom(variable);
+                environmentVariables.add(environmentVariable);
+            }
+
+            setEnvironmentVariables(environmentVariables);
+        }
+
+        if (model.registryCredential() != null) {
+            CodebuildRegistryCredential registryCredential = newSubresource(CodebuildRegistryCredential.class);
+            registryCredential.copyFrom(model.registryCredential());
+            setRegistryCredential(registryCredential);
+        }
+    }
+
+    @Override
+    public String primaryKey() {
+        return "";
+    }
+
+    @Override
+    public List<ValidationError> validate(Set<String> configuredFields) {
+        List<ValidationError> errors = new ArrayList<>();
+
+        if (getRegistryCredential() != null && !getImagePullCredentialsType().equals("SERVICE_ROLE")) {
+            errors.add(new ValidationError(
+                this,
+                null,
```

Comment:
Even though the validation is based on two fields, this validation is pertaining to `registry-credential` 
```suggestion
               "registry-credential",
```